sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -919,49 +974,11 @@ static void _receive(gnrc_pktsnip_t *pkt)
             return;
         }
         /* TODO: check if receiving interface is router */
-        else if (--(hdr->hl) > 0) {  /* drop packets that *reach* Hop Limit 0 */
-            gnrc_pktsnip_t *reversed_pkt = NULL, *ptr = pkt;
-
-            DEBUG("ipv6: forward packet to next hop\n");
-
-            /* pkt might not be writable yet, if header was given above */
-            ipv6 = gnrc_pktbuf_start_write(ipv6);
-            if (ipv6 == NULL) {
-                DEBUG("ipv6: unable to get write access to packet: dropping it\n");
-                gnrc_pktbuf_release(pkt);
-                return;
-            }
-
-            /* remove L2 headers around IPV6 */
-            netif = gnrc_pktsnip_search_type(pkt, GNRC_NETTYPE_NETIF);
-            if (netif != NULL) {
-                gnrc_pktbuf_remove_snip(pkt, netif);
-            }
-
-            /* reverse packet snip list order */
-            while (ptr != NULL) {
-                gnrc_pktsnip_t *next;
-                ptr = gnrc_pktbuf_start_write(ptr);     /* duplicate if not already done */
-                if (ptr == NULL) {
-                    DEBUG("ipv6: unable to get write access to packet: dropping it\n");
-                    gnrc_pktbuf_release(reversed_pkt);
-                    gnrc_pktbuf_release(pkt);
-                    return;
-                }
-                next = ptr->next;
-                ptr->next = reversed_pkt;
-                reversed_pkt = ptr;
-                ptr = next;
-            }
-            _send(reversed_pkt, false);
-            return;
-        }
-        else {
```

Comment:
I migrated the responsibility back to IPv6 [1]. 
The actual forwarding of the packet is now performed when the demultiplexing is finished [2].


[1] https://github.com/RIOT-OS/RIOT/pull/7231/files#diff-74976d45a42eaa063e3561b5fd36f615R146
[2] https://github.com/RIOT-OS/RIOT/pull/7231/files#diff-74976d45a42eaa063e3561b5fd36f615R255